File: 2nd_quarter/week03/whaeun/삼총사.cpp
#include <string>
#include <vector>

using namespace std;

int answer = 0;

void generateCombinations(vector<int>& nums, vector<int>& currComb, int index){
    if (currComb.size() == 3) {
        int sum = 0;
        for (int num : currComb) {
            sum += num;
        }
        
        if(sum == 0){
            answer++;
        }
        return;
    }
    
    if(index == nums.size()){
        return;
    }
    
    currComb.push_back(nums[index]);
    generateCombinations(nums, currComb, index + 1);
    currComb.pop_back();
    
    generateCombinations(nums, currComb, index + 1);
}


int solution(vector<int> number) {
    
    vector<int> currComb;
    
    generateCombinations(number, currComb, 0);
    
    return answer;
}

File: 2nd_quarter/week03/Jose/삼총사.py
# 오 .. 무슨 알고리즘이 있으려나 ???
# 가능한 수를 만들라고 했으니 - 
# 예아 좋다 ! 
 
def solution(number):
    answer = 0
    
    length = len(number)
    
    temp = []
    
    for i in range(length):
        temp.append(number[i])
        
        for j in range(i+1, length):
            temp.append(number[j])
            
            for k in range(j+1, length):
                temp.append(number[k])
                if sum(temp) == 0:
                    answer += 1
                temp.pop()
            temp.pop()
        temp.pop()
    return answer
